ci_framework/playbooks/07-deploy-edpm-with-bmaas.yml

```diff
@@ -0,0 +1,46 @@
+- hosts: "{{ cifmw_target_host | default('localhost') }}"
+  gather_facts: false
+  tasks:
+    - name: Call make_edpm_compute_baremetal target
+      ansible.builtin.include_role:
+        name: 'install_yamls_makes'
+        tasks_from: 'make_edpm_compute_baremetal'
+      vars:
+        make_edpm_compute_baremetal_env:
+          NETWORK_ISOLATION: false
+          OPENSTACK_IMG: "{{ cifmw_operator_build_output['operators']['openstack-operator']['image_catalog'] }}"
```

Comment:
Not sure I understand fully the context of the question. However, if the question is `why we need to be build openstack-operator for PRs?`, both dataplane-operator and openstack-baremetal-operator are bundled in openstack-operator, so we need to build that.

Reply:
@pablintino

Based on my initial conversation with @raukadah, my understanding is that we need to consider two types of scenarios:

1) When running the job against actual repositories like dataplane-operator and openstack-baremetal-operator, we need to build the actual operator and meta-operator images using the content-provider job. In this case, the meta_operator IMG information is passed from the content-provider to the actual job through the cifmw_operator_build_output variable.


https://logserver.rdoproject.org/08/208/619dfcd11a446724938146462d2ba9ec4a3a0cbe/github-check/dataplane-operator-crc-podified-edpm-deployment/fee2a1d/zuul-info/inventory.yaml
~~~
      cifmw_operator_build_output:
        operators:
          openstack-operator::
.
.
            image_catalog: 38.102.83.181:5001/openstack-k8s-operators/openstack-operator-index:7a4dc9906a69ab2dd7937dc5819c2c7b4855b69b
~~~


2)  On the other hand, when running the job against repositories like install_yamls and ci-framework, there is no need to build operator images. Instead, we can pull the latest image. For this scenario, we have followed a pattern where a child job is created and the latest image is defined using the following variables.

https://github.com/openstack-k8s-operators/ci-framework/blob/main/zuul.d/edpm.yaml#L41-L55
~~~
    vars:
      cifmw_operator_build_output:
          operators:
              openstack-operator:
                  git_src_dir: /home/zuul-worker/src/github.com/openstack-k8s-operators/openstack-operator
                  image_catalog: quay.io/openstack-k8s-operators/openstack-operator-index:latest
~~~  

Therefore, in cases where we don't need to build meta-operators, we are covered because we have defined the latest image using the job variable.


Please let me know if you have any further suggestions or if you need any additional information.

Reply:
Well, honestly, `cifmw_operator_build_output` wasn't meant to be used as an input variable, the input variable of the container build step is `cifmw_operator_build_operators`, passing the `cifmw_operator_build_output` will probably break the operator_build as -e passed variables will make the role not able to set the proper value of `cifmw_operator_build_output` in steps like https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/operator_build/tasks/build.yml#L97-L107
We will need a proper way to handle your use case. @raukadah be aware if the code that we already have this way, the one that @Sandeepyadav93 suggests https://logserver.rdoproject.org/08/208/619dfcd11a446724938146462d2ba9ec4a3a0cbe/github-check/dataplane-operator-crc-podified-edpm-deployment/fee2a1d/zuul-info/inventory.yaml, this only works if 'cifmw_operator_build_operators' is empty, if I remember correctly, as a variable with that name exists, setting a fact with the same name or altering it does nothing..., making the previous lines I posted useless.

Reply:
@rabi @Sandeepyadav93 In the other hand, with that comment I was refering to not expect `cifmw_operator_build_output` to be populated, in local-envs where nothing is build an the latest images form quay are fetched that variable is not defined/is empty, I now that your use case is something that maybe is not locally testable, but maybe is a good a idea to properly handle accessing this dict if we implement the steps skipping logic, that will made possible to have this dict empty. Obviously, if you assume that you are gonna pass it always externally that comment may not apply, but as discussed in the previous comments, I don't think that passing the 05 step output variable is the way to go, seems more like a hack than a proper wy of handle this situation.

Reply:
@Sandeepyadav93 In the existing crc job, we do `make openstack` in the config repo crc playbook. In order to make them work with both normal and content provider dependent job, I have added this var at both places:
1. In[ ci-framework zuul.d edpm.yaml](https://github.com/openstack-k8s-operators/ci-framework/blob/main/zuul.d/edpm.yaml#L51)
2. in Config repo, [crc openstack playbook](https://github.com/rdo-infra/review.rdoproject.org-config/blob/master/playbooks/crc/openstack.yaml#L40)

As we move towards the `base-simple-crc` parent we do not need these vars in the job definition as CI-framework edpm_prepare role can take of setting the proper vars.